module.c

```diff
@@ -442,6 +509,77 @@ static PyObject *context_gc(ContextData *self) {
 	Py_RETURN_NONE;
 }
 
+static JSValue js_c_function(
+    JSContext *ctx, JSValueConst this_val, int argc, JSValueConst *argv, int magic) {
+	ContextData *self = (ContextData *)JS_GetContextOpaque(ctx);
+	PythonCallableNode *node = self->python_callables;
+	while (node && node->magic != magic) {
+		node = node->next;
+	}
+	if (!node) {
+		return JS_ThrowInternalError(self->context, "Internal error.");
+	}
+
+	PyObject *args = PyTuple_New(argc);
+	if (!args) {
+		return JS_ThrowOutOfMemory(self->context);
+	}
+	for (int i = 0; i < argc; ++i) {
+		PyObject *arg = quickjs_to_python(self, JS_DupValue(self->context, argv[i]));
+		PyTuple_SET_ITEM(args, i, arg);
+	}
+
+	PyObject *result = PyObject_CallObject(node->obj, args);
+	Py_DECREF(args);
+	if (!result) {
+		return JS_ThrowInternalError(self->context, "Python call failed.");
+	}
+	JSValue js_result = JS_NULL;
+	if (python_to_quickjs_possible(self, result)) {
+		js_result = python_to_quickjs(self, result);
+	}
```

Comment:
This API is kinda silly, how about calling python_to_quickjs and checking for error (e.g. JS_IsUndefined or an explicit outparam)?

Reply:
This API simplifies the memory management in `object_call`.